### PR TITLE
Remove references to PlatformAbstractions.

### PR DESCRIPTION
**Pull request by eerhardt**
This will allow us to remove the PlatformAbstractions library from dotnet/runtime.

I needed to update the SDK used so I can use the new RuntimeInformation.RuntimeIdentifier API that is only in .NET 5.

Contributes to https://github.com/dotnet/runtime/issues/3470

This emulates a similar change in dotnet/runtime: https://github.com/dotnet/runtime/commit/a1af15d229c44f72d639c686a66a621196fd4172

cc @NikolaMilosavljevic @dagood @AdamYoblick

### CODE REVIEW

**Comment by AdamYoblick** [comment]
@vatsan-madhavan I'm not sure who to tag on this from WPF. Just wanted to make sure that moving windowsdesktop repo to .net 5 won't break anything from wpf?

File: eng/Signing.props [hunk 1]
```diff
@@ -23,21 +23,6 @@
...
 
-  <ItemGroup Condition="'$(SignBinaries)' == 'true'">
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/hostfxr.dll" />
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/hostpolicy.dll" />
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/dotnet.exe" />
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/ijwhost.dll" />
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/winrthost.dll" />
-    <ItemsToSign Include="$(BaseOutputRootPath)corehost/**/nethost.dll" />
-
-    <ItemsToSign Include="$(CrossGenRootPath)**/*.dll" />
```
Comment (dagood) [inline eng/Signing.props:34]: The surrounding line deletes look like good cleanup, but I don't understand this one. Isn't this needed to sign the crossgen'd WindowsDesktop bits?

**Comment by vatsan-madhavan** [comment]
@AdamYoblick, please tag @dotnet/wpf-developers 😄 

I worry that it has the potential to break something in our tests... but maybe it's benign after all. I can't answer this without research unfortunately. 

@fabiant3, can someone please vet this change and signoff ?

File: eng/Signing.props (hunk 1 above)
Comment (eerhardt) [reply eng/Signing.props:34]: I might have been too aggressive here, as the other lines were duplicated from "core-setup".

But I did search for someone setting `$(SignBinaries)`, and I couldn't find it being used. Did I miss it somewhere?

Comment (dagood) [reply eng/Signing.props:34]: Easy to miss:

https://github.com/dotnet/windowsdesktop/blob/b8893fd3e27cc169ee519629f191dc8336b2b423/signing/Directory.Build.targets#L8-L17

https://github.com/dotnet/windowsdesktop/blob/master/signing/SignBinaries.proj

**Comment by eerhardt** [comment]
Any thoughts/feedback on this? `PlatformAbstractions` is about to be removed.

**Comment by fabiant3** [comment]
@ryalanms for awareness.

**Comment by mmitche** [comment]
I think we need this to go into master and then be cherry-picked to p5.

/cc @lukas-lansky

**Comment by mmitche** [comment]
@AdamYoblick Can this be merged?

**Comment by AdamYoblick** [comment]
I think so? I'm really not familiar enough with this repo to say for sure. Also, someone from wpf should also sign off since this is a shared packaging repo.

**Comment by mmitche** [comment]
Yeah. I think this is blocking p5 at the moment. @vatsan-madhavan @eerhardt thoughts on merging this?

**Comment by eerhardt** [comment]
I have no objections to merging this, but there aren't any approvals yet.

**Comment by AdamYoblick** [comment]
I've approved for winforms, but @ryalanms (or someone else) should probably approve for wpf. 😄

**Comment by vatsan-madhavan** [comment]
I take back my previous comment about potential for breaking (wpf) tests etc. I really can't tell without research; I also don't have a strong reason to think that this _would_ cause problems. Please proceed as if I didn't say anything in the first place 😀. 

I usually like to give these sorts of changes a thorough once-over and understand (a) impact to the product and (b) impact to immediate (build-time or run-time) dependencies and (c) impact to customers. There is just not enough information in the PR to make any of those judgments easily; (a) is sort of cryptic ("go read something in dotnet/runtime" is what I'm seeing... "remove PlatformAbstractions".. but why?), (b) requires at least a basic bit of analysis of closely related repos (like wpf*, winforms etc.) - probably no impact there but good to make sure nothing from there relies on content from "PlatformAbstractions", and I'm not sure what's needed for (c). I don't have the time to go digging any further -deferring to @fabiant3 .

**Comment by eerhardt** [comment]
> (a) impact to the product 

There shouldn't be any impact to the product, this solely a build infrastructure change.

> (b) impact to immediate (build-time or run-time) dependencies

This removes a build-time dependency on the [Microsoft.DotNet.PlatformAbstractions](https://www.nuget.org/packages/Microsoft.DotNet.PlatformAbstractions) library. The reason it is being removed is because the library is being obsoleted.

This change has no impact on run-time dependencies.

> (c) impact to customers

There shouldn't be any impact to customers, this solely a build infrastructure change.

**Comment by vatsan-madhavan** [comment]
Was it used in this repo purely to power the build-task functionality in GetTargetMachineInfo.cs ? If yes, and if it wasn't present to support any other purpose related to winforms or wpf, then this change seems reasonable to me.

**Comment by eerhardt** [comment]
> Was it used in this repo purely to power the build-task functionality in GetTargetMachineInfo.cs ?

Correct. Previously, `PlatformAbstractions` was the only way to get the current `RuntimeIdentifier` of the machine. The only thing used from `GetTargetMachineInfo.cs` was to get the "host RuntimeIdentifier", which I replaced with the new API: `$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)`.

**Pull request merged**